Client/Frontend/Browser/BrowserViewController/OpenSearch/OpenSearchEngineButton.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import SnapKit
+import BraveShared
+
+// MARK: - OpenSearchEngineButton
+
+class OpenSearchEngineButton: UIView {
+
+    // MARK: State
+    
+    enum State {
+        case loading
+        case enabled
+        case disabled
+    }
+
+    // MARK: Properties
+    
+    var state: State {
+        didSet {
+            switch state {
+            case .disabled:
+                searchButton.isHidden = hidesWhenDisabled ? true : false
+                loadingIndicator.stopAnimating()
+                searchButton.tintColor = UIColor.Photon.grey50
+                searchButton.setTitleColor(UIColor.Photon.grey50, for: .normal)
+                searchButton.isUserInteractionEnabled = false
+            case .enabled:
+                searchButton.isHidden = false
+                loadingIndicator.stopAnimating()
+                searchButton.tintColor = .systemBlue
+                searchButton.setTitleColor(BraveUX.braveOrange, for: .normal)
+                searchButton.isUserInteractionEnabled = true
+            case .loading:
+                loadingIndicator.isHidden = false
+                loadingIndicator.startAnimating()
+                searchButton.isHidden = true
+            }
+        }
+    }
+    
+    private let searchButton = UIButton().then {
+        $0.setImage(#imageLiteral(resourceName: "AddSearch").template, for: [])
+        $0.titleLabel?.font = UIFont.systemFont(ofSize: 14)
```

Comment:
Use `.preferredFontSize(for: .body)` instead?

Reply:
Preferred Font Size is used instead where constant 14.0 system font size defined in various places. Fix at https://github.com/brave/brave-ios/pull/3231/commits/ba493be2def981cae70ab0ce8231326645c31a0d